pandas/core/arrays/datetimelike.py

```diff
@@ -520,7 +521,11 @@ def __getitem__(self, key):
         if com.is_bool_indexer(key):
             # first convert to boolean, because check_array_indexer doesn't
             # allow object dtype
-            key = np.asarray(key, dtype=bool)
+            if is_extension_array_dtype(key):
```

Comment:
hmm, not sure you actually need to do this here; as you already patched check_array_indexer which does this.